src/domain/swaps/contracts/decoders/composable-cow-decoder.helper.ts

```diff
@@ -0,0 +1,269 @@
+   * @param staticInput - IConditionalOrder.ConditionalOrderParams calldata
+   *
+   * @returns {@link TwapData} decoded from the static input
+   */
+  private decodeConditionalOrderParams(
+    staticInput: `0x${string}`, // IConditionalOrder.ConditionalOrderParams calldata
+  ): TwapData {
+    const [
+      sellToken,
+      buyToken,
+      receiver,
+      partSellAmount,
+      minPartLimit,
+      startEpoch, // t0
+      numberOfParts, // n
+      timeBetweenParts, // t
+      span,
+      appData,
+    ] = decodeAbiParameters(
+      ComposableCowDecoder.TwapStructAbiParameters,
+      staticInput,
+    );
+
+    const isSpanZero = span === BigInt(0);
+
+    const durationOfPart: DurationOfPart = isSpanZero
+      ? { durationType: DurationType.AUTO }
+      : { durationType: DurationType.LIMIT_DURATION, duration: span };
+
+    const startTime: StartTime = isSpanZero
+      ? { startType: StartTimeValue.AT_MINING_TIME }
+      : { startType: StartTimeValue.AT_EPOCH, epoch: startEpoch };
+
+    return {
+      sellToken,
+      buyToken,
+      receiver,
+      sellAmount: partSellAmount * numberOfParts,
+      buyAmount: minPartLimit * numberOfParts,
+      startTime,
+      numberOfParts,
+      timeBetweenParts,
+      durationOfPart,
+      appData, // TODO: Decode?
+    };
+  }
+}
+
+interface TwapData {
+  /**
+   * Meta-data associated with the order. Normally would be the keccak256 hash of the document generated in http://github.com/cowprotocol/app-data
+   *
+   * This hash should have been uploaded to the API https://api.cow.fi/docs/#/default/put_api_v1_app_data__app_data_hash_ and potentially to other data availability protocols like IPFS.
+   *
+   */
+  readonly appData: string;
+
+  /**
+   * minimum amount of buyToken that must be bought across the entire TWAP
+   */
+  readonly buyAmount: bigint;
+
+  /**
+   * which token to buy
+   */
+  readonly buyToken: string;
+
+  /**
+   * whether the TWAP is valid for the entire interval or not
+   */
+  readonly durationOfPart?: DurationOfPart;
+
+  /**
+   * number of parts
+   */
+  readonly numberOfParts: bigint;
+
+  /**
+   * who to send the tokens to
+   */
+  readonly receiver: string;
+
+  /**
+   * total amount of sellToken to sell across the entire TWAP
+   */
+  readonly sellAmount: bigint;
+
+  /**
+   * which token to sell
+   */
+  readonly sellToken: string;
+
+  /**
+   * start time of the TWAP
+   */
+  readonly startTime?: StartTime;
+
+  /**
+   * duration of the TWAP interval
+   */
+  readonly timeBetweenParts: bigint;
+}
+
+type DurationOfPart =
+  | { durationType: DurationType.AUTO }
+  | { durationType: DurationType.LIMIT_DURATION; duration: bigint };
+
+enum DurationType {
+  AUTO = 'AUTO',
+  LIMIT_DURATION = 'LIMIT_DURATION',
+}
+
+type StartTime =
+  | { startType: StartTimeValue.AT_MINING_TIME }
+  | { startType: StartTimeValue.AT_EPOCH; epoch: bigint };
+
+enum StartTimeValue {
+  AT_MINING_TIME = 'AT_MINING_TIME',
+  AT_EPOCH = 'AT_EPOCH',
+}
```

Comment:
I intend to eventually move these entities but will first wait for a pattern to develop as the relative mapper may inherit some of this.